perf/reader.go

```diff
@@ -215,6 +216,9 @@ func NewReaderWithOptions(array *ebpf.Map, perCPUBuffer int, opts ReaderOptions)
 	for i := 0; i < nCPU; i++ {
 		ring, err := newPerfEventRing(i, perCPUBuffer, opts.Watermark)
 		if err != nil {
+			if xerrors.Is(err, syscall.ENODEV) {
```

Comment:
The code expects you to append to `rings` for each CPU, and right now you skip this because of the continue. I need to look into how to handle that.